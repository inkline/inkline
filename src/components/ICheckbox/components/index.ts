export { default as ICheckboxGroup } from './ICheckboxGroup/ICheckboxGroup.vue';
