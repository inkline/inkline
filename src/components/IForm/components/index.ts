export { default as IFormGroup } from './IFormGroup/iFormGroup.vue';
export { default as IFormError } from './IFormError/IFormError.vue';
export { default as IFormLabel } from './IFormLabel/IFormLabel.vue';
