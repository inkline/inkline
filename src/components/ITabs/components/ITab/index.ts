export { default as ITab } from './ITab.vue';