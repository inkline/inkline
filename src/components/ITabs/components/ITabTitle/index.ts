export { default as ITabTitle } from './ITabTitle.vue';