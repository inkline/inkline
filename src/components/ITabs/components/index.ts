export { default as ITab } from '@inkline/inkline/components/ITabs/components/ITab/index.vue';
export { default as ITabTitle } from '@inkline/inkline/components/ITabs/components/ITabTitle/index.vue';
