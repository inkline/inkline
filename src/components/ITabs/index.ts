export { default as ITabs } from './ITabs.vue';