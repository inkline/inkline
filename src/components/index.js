export { default as IAlert } from '@inkline/inkline/src/components/Alert';
export { default as IBadge } from '@inkline/inkline/src/components/Badge';
export { default as IBreadcrumb } from '@inkline/inkline/src/components/Breadcrumb';
export { default as IBreadcrumbItem } from '@inkline/inkline/src/components/BreadcrumbItem';
export { default as IButton } from '@inkline/inkline/src/components/Button';
export { default as IButtonGroup } from '@inkline/inkline/src/components/ButtonGroup';
export { default as ICard } from '@inkline/inkline/src/components/Card';
export { default as ICollapsible } from '@inkline/inkline/src/components/Collapsible';
export { default as ICollapsibleItem } from '@inkline/inkline/src/components/CollapsibleItem';
export { default as IColumn } from '@inkline/inkline/src/components/Column';
export { default as IContainer } from '@inkline/inkline/src/components/Container';
export { default as IDropdown } from '@inkline/inkline/src/components/Dropdown';
export { default as IDropdownDivider } from '@inkline/inkline/src/components/DropdownDivider';
export { default as IDropdownItem } from '@inkline/inkline/src/components/DropdownItem';
export { default as IDropdownMenu } from '@inkline/inkline/src/components/DropdownMenu';
export { default as IHamburgerMenu } from '@inkline/inkline/src/components/HamburgerMenu';
export { default as IHeader } from '@inkline/inkline/src/components/Header';
export { default as IIcon } from '@inkline/inkline/src/components/Icon';
export { default as ILayout } from '@inkline/inkline/src/components/Layout';
export { default as ILayoutAside } from '@inkline/inkline/src/components/LayoutAside';
export { default as ILayoutContent } from '@inkline/inkline/src/components/LayoutContent';
export { default as ILayoutFooter } from '@inkline/inkline/src/components/LayoutFooter';
export { default as ILayoutHeader } from '@inkline/inkline/src/components/LayoutHeader';
export { default as IListGroup } from '@inkline/inkline/src/components/ListGroup';
export { default as IListGroupItem } from '@inkline/inkline/src/components/ListGroupItem';
export { default as ILoader } from '@inkline/inkline/src/components/Loader';
export { default as IMedia } from '@inkline/inkline/src/components/Media';
export { default as IModal } from '@inkline/inkline/src/components/Modal';
export { default as INav } from '@inkline/inkline/src/components/Nav';
export { default as INavItem } from '@inkline/inkline/src/components/NavItem';
export { default as INavbar } from '@inkline/inkline/src/components/Navbar';
export { default as INavbarBrand } from '@inkline/inkline/src/components/NavbarBrand';
export { default as INavbarItems } from '@inkline/inkline/src/components/NavbarItems';
export { default as IRow } from '@inkline/inkline/src/components/Row';
export { default as IPagination } from '@inkline/inkline/src/components/Pagination';
export { default as IPopover } from '@inkline/inkline/src/components/Popover';
export { default as IProgress } from '@inkline/inkline/src/components/Progress';
export { default as IProgressBar } from '@inkline/inkline/src/components/ProgressBar';
export { default as ITable } from '@inkline/inkline/src/components/Table';
export { default as ITooltip } from '@inkline/inkline/src/components/Tooltip';

export { default as ICheckbox } from '@inkline/inkline/src/components/Checkbox';
export { default as ICheckboxButton } from '@inkline/inkline/src/components/CheckboxButton';
export { default as ICheckboxGroup } from '@inkline/inkline/src/components/CheckboxGroup';
export { default as ICheckboxButtonGroup } from '@inkline/inkline/src/components/CheckboxButtonGroup';
export { default as IInput } from '@inkline/inkline/src/components/Input';
export { default as IForm } from '@inkline/inkline/src/components/Form';
export { default as IFormGroup } from '@inkline/inkline/src/components/FormGroup';
export { default as IFormLabel } from '@inkline/inkline/src/components/FormLabel';
export { default as IInputNumber } from '@inkline/inkline/src/components/InputNumber';
export { default as IRadio } from '@inkline/inkline/src/components/Radio';
export { default as IRadioButton } from '@inkline/inkline/src/components/RadioButton';
export { default as IRadioGroup } from '@inkline/inkline/src/components/RadioGroup';
export { default as IRadioButtonGroup } from '@inkline/inkline/src/components/RadioButtonGroup';
export { default as ISelect } from '@inkline/inkline/src/components/Select';
export { default as ISelectOption } from '@inkline/inkline/src/components/SelectOption';
export { default as ITextarea } from '@inkline/inkline/src/components/Textarea';
