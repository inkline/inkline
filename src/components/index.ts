export { default as IAlert } from '@inkline/inkline/components/IAlert/index.vue';
export { default as IBadge } from '@inkline/inkline/components/IBadge/index.vue';
export { default as IBreadcrumb } from '@inkline/inkline/components/IBreadcrumb/index.vue';
export * from '@inkline/inkline/components/IBreadcrumb/components';
export { default as IButton } from '@inkline/inkline/components/IButton/index.vue';
export { default as IButtonGroup } from '@inkline/inkline/components/IButtonGroup/index.vue';
export { default as IColumn } from '@inkline/inkline/components/IColumn/index.vue';
export { default as IContainer } from '@inkline/inkline/components/IContainer/index.vue';
export { default as ICard } from '@inkline/inkline/components/ICard/index.vue';
export { default as ICheckbox } from '@inkline/inkline/components/ICheckbox/index.vue';
export * from '@inkline/inkline/components/ICheckbox/components';
export { default as ICollapsible } from '@inkline/inkline/components/ICollapsible/index.vue';
export * from '@inkline/inkline/components/ICollapsible/components';
export { default as IDropdown } from '@inkline/inkline/components/IDropdown/index.vue';
export * from '@inkline/inkline/components/IDropdown/components';
export { default as IForm } from '@inkline/inkline/components/IForm/index.vue';
export * from '@inkline/inkline/components/IForm/components';
export { default as IHamburgerMenu } from '@inkline/inkline/components/IHamburgerMenu/index.vue';
export { default as IHeader } from '@inkline/inkline/components/IHeader/index.vue';
export { default as IIcon } from '@inkline/inkline/components/IIcon/index.vue';
export { default as IInput } from '@inkline/inkline/components/IInput/index.vue';
export { default as INumberInput } from '@inkline/inkline/components/INumberInput/index.vue';
export { default as ITextarea } from '@inkline/inkline/components/ITextarea/index.vue';
export { default as ILayout } from '@inkline/inkline/components/ILayout/index.vue';
export { default as ILayoutAside } from '@inkline/inkline/components/ILayout/components/ILayoutAside/index.vue';
export { default as ILayoutContent } from '@inkline/inkline/components/ILayout/components/ILayoutContent/index.vue';
export { default as ILayoutFooter } from '@inkline/inkline/components/ILayout/components/ILayoutFooter/index.vue';
export { default as ILayoutHeader } from '@inkline/inkline/components/ILayout/components/ILayoutHeader/index.vue';
export { default as IListGroup } from '@inkline/inkline/components/IListGroup/index.vue';
export * from '@inkline/inkline/components/IListGroup/components';
export { default as ILoader } from '@inkline/inkline/components/ILoader/index.vue';
export { default as IMedia } from '@inkline/inkline/components/IMedia/index.vue';
export { default as IModal } from '@inkline/inkline/components/IModal/index.vue';
export { default as INav } from '@inkline/inkline/components/INav/index.vue';
export * from '@inkline/inkline/components/INav/components';
export { default as INavbar } from '@inkline/inkline/components/INavbar/index.vue';
export * from '@inkline/inkline/components/INavbar/components';
export { default as IPagination } from '@inkline/inkline/components/IPagination/index.vue';
export { default as IPopover } from '@inkline/inkline/components/IPopover/index.vue';
export { default as IProgress } from '@inkline/inkline/components/IProgress/index.vue';
export * from '@inkline/inkline/components/IProgress/components';
export { default as IRadio } from '@inkline/inkline/components/IRadio/index.vue';
export * from '@inkline/inkline/components/IRadio/components';
export { default as IRow } from '@inkline/inkline/components/IRow/index.vue';
export { default as ISelect } from '@inkline/inkline/components/ISelect/index.vue';
export * from '@inkline/inkline/components/ISelect/components';
export { default as ISidebar } from '@inkline/inkline/components/ISidebar/index.vue';
export { default as ITable } from '@inkline/inkline/components/ITable/index.vue';
export { default as Tabs } from '@inkline/inkline/components/ITabs/index.vue';
export * from '@inkline/inkline/components/ITabs/components';
export { default as IToggle } from '@inkline/inkline/components/IToggle/index.vue';
export { default as ITooltip } from '@inkline/inkline/components/ITooltip/index.vue';
