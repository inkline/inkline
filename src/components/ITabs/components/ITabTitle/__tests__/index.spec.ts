import { fireEvent, render } from '@testing-library/vue';
import { ITabTitle } from '@inkline/inkline/components';

describe('Components', () => {
    describe('ITabTitle', () => {
        const props = {
            for: 'tab'
        };

        it('should be named correctly', () => {
            expect(ITabTitle.name).toEqual('ITabTitle');
        });

        it('should render correctly', () => {
            const wrapper = render(ITabTitle, { props });

            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('props', () => {
            describe('for', () => {
                it('should have randomly generated name uid', () => {
                    const wrapper = render(ITabTitle);

                    expect(wrapper.container.firstChild).toHaveAttribute('for', expect.stringContaining('tab'));
                });
            });
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should have \'-active\' class if active', () => {
                    const wrapper = render(ITabTitle, {
                        global: {
                            provide: {
                                tabs: {
                                    active: 'tab'
                                }
                            }
                        },
                        props
                    });

                    expect(wrapper.container.firstChild).toHaveClass(
                        '-active'
                    );
                });

                it('should not have \'-active\' class if not active', () => {
                    const wrapper = render(ITabTitle, {
                        global: {
                            provide: {
                                tabs: {
                                    active: 'other'
                                }
                            }
                        },
                        props
                    });

                    expect(wrapper.container.firstChild).not.toHaveClass(
                        '-active'
                    );
                });
            });
        });

        describe('methods', () => {
            describe('onClick()', () => {
                it('should have call tabs setActive', async () => {
                    const setActive = vi.fn();
                    const wrapper = render(ITabTitle, {
                        global: {
                            provide: {
                                tabs: {
                                    setActive
                                }
                            }
                        },
                        props
                    });

                    await fireEvent.click(wrapper.container.firstChild as Element);

                    expect(setActive).toHaveBeenCalledWith(props.for);
                });
            });
        });
    });
});
