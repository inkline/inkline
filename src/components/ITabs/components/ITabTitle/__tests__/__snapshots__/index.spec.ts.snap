// Vitest Snapshot v1

exports[`Components > ITabTitle > should render correctly 1`] = `"<div class=\\"tab-title\\" role=\\"tab\\" for=\\"tab\\" active=\\"false\\" aria-expanded=\\"false\\" aria-controls=\\"tab-content-tab\\" aria-describedby=\\"tab-content-tab\\" tabindex=\\"0\\"></div>"`;
